src/core/service.c

```diff
@@ -1561,21 +1561,43 @@ static int service_spawn(
                 }
         }
 
-        if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
-                        return -ENOMEM;
+        if (flags & (EXEC_SETENV_RESULT | EXEC_SETENV_MONITOR_RESULT)) {
```

Comment:
I think indentation could be reduced by merging this if and `if (env_source)` - move the assert and the declaration of env_source outside it and merge the conditions?

Reply:
yep,sounds good.